<div class="container">

<table style="width: 100%;"><tr>
<td>slap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Slap Operator</h2>

<h3>Description</h3>

<p>Slap Operator
</p>


<h3>Usage</h3>

<pre><code class="language-R">expr %!% message

expr %!!% message
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>expr</code></td>
<td>
<p>An expression or quosure to evaluate carefully</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>message</code></td>
<td>
<p>A message meant to be formatted by <code>cli::cli_bullets()</code> or a
function.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>expr</code> succeeds, its result is returned.
</p>
<p>When <code>expr</code> generates an error, the <code style="white-space: pre;">⁠%!%⁠</code> and <code style="white-space: pre;">⁠%!!%⁠</code> operators
catch it and embed it in a new error thrown by <code>cli::cli_abort()</code>.
</p>
<p>If <code>message</code> evaluates to a character vector, it is used as the
<code>message</code> argument of <code>cli::cli_abort()</code>.
</p>
<p>If <code>message</code> evaluates to a function, the function is called with one
argument: the caught error from evaluating <code>expr</code>.
</p>
<p>When the current environment has an <code>error_call</code> object, it is
used as the <code>call</code> argument of <code>cli::cli_abort()</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># g() throws an error
g &lt;- function() {
  stop("ouch")
}

# h() catches that error and embed it in a new error
# with "bam" as its message, the g() error as the parent error,
# and the caller environment as call=
h &lt;- function(error_call = rlang::caller_env()) {
  g() %!% "bam"
}

# f() will be used as the error call
f &lt;- function() {
  h()
}

# Error in `f()`:
# ! bam
# Caused by error in `g()`:
# ! ouch
tryCatch(f(), error = function(err) {
  print(err, backtrace = FALSE)
})

</code></pre>


</div>